Mathlib/RingTheory/Polynomial/Hermite/Basic.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle
+
+! This file was ported from Lean 3 source module ring_theory.polynomial.hermite.basic
+! leanprover-community/mathlib commit 066ecdb4834c7a4693e0f0e5154935a6f3d3f90c
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Polynomial.Derivative
+import Mathlib.Data.Nat.Parity
+
+/-!
+# Hermite polynomials
+
+This file defines `Polynomial.hermite n`, the nth probabilist's Hermite polynomial.
+
+## Main definitions
+
+* `Polynomial.hermite n`: the `n`th probabilist's Hermite polynomial,
+  defined recursively as a `Polynomial ℤ`
+
+## Results
+
+* `Polynomial.hermite_succ`: the recursion `hermite (n+1) = (x - d/dx) (hermite n)`
+* `Polynomial.coeff_hermite_of_odd_add`: for `n`,`k` where `n+k` is odd, `(hermite n).coeff k` is
+  zero.
+* `Polynomial.monic_hermite`: for all `n`, `hermite n` is monic.
+
+## References
+
+* [Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+
+noncomputable section
+
+open Polynomial
+
+namespace Polynomial
+
+/-- the nth probabilist's Hermite polynomial -/
+noncomputable def hermite : ℕ → Polynomial ℤ
+  | 0 => 1
+  | n + 1 => X * hermite n - Polynomial.derivative (hermite n)
+#align polynomial.hermite Polynomial.hermite
+
+/-- The recursion `hermite (n+1) = (x - d/dx) (hermite n)` -/
+@[simp]
+theorem hermite_succ (n : ℕ) :
+    hermite (n + 1) = X * hermite n - Polynomial.derivative (hermite n) := by
+  rw [hermite]
+#align polynomial.hermite_succ Polynomial.hermite_succ
+
+theorem hermite_eq_iterate (n : ℕ) :
+    hermite n = ((fun p => X * p - Polynomial.derivative p)^[n]) 1 := by
+  induction' n with n ih
+  · rfl
+  · rw [Function.iterate_succ_apply', ← ih, hermite_succ]
+#align polynomial.hermite_eq_iterate Polynomial.hermite_eq_iterate
+
+@[simp]
+theorem hermite_zero : hermite 0 = C 1 :=
+  rfl
+#align polynomial.hermite_zero Polynomial.hermite_zero
+
+-- Porting note: There was initially @[simp] on this line but it was removed
+-- because simp can prove this theorem
+theorem hermite_one : hermite 1 = X := by
+  rw [hermite_succ, hermite_zero]
+  simp only [map_one, mul_one, derivative_one, sub_zero]
+#align polynomial.hermite_one Polynomial.hermite_one
+
+/-! ### Lemmas about `Polynomial.coeff` -/
+
+
+section Coeff
+
+theorem coeff_hermite_succ_zero (n : ℕ) : coeff (hermite (n + 1)) 0 = -coeff (hermite n) 1 := by
+  simp [coeff_derivative]
+#align polynomial.coeff_hermite_succ_zero Polynomial.coeff_hermite_succ_zero
+
+theorem coeff_hermite_succ_succ (n k : ℕ) :
+  coeff (hermite (n + 1)) (k + 1) = coeff (hermite n) k - (k + 2) * coeff (hermite n) (k + 2) := by
```

Comment:
```suggestion
theorem coeff_hermite_succ_succ (n k : ℕ) : coeff (hermite (n + 1)) (k + 1) = 
    coeff (hermite n) k - (k + 2) * coeff (hermite n) (k + 2) := by
```
Unfortunately the Lean 4 spacing convention requires 4 spaces on a multiline theorem, this takes us just over the 100 line limit. The other option is to move `by` but this would also go against a Lean 4 convention :-/